### ISSUE TITLE
'h' format requires -32768 <= number <= 32767

### ISSUE THREAD

**Issue by rnd-rivertech** [lkempf/casambi-bt-hass#28]
Hi there. Great job. Some of my lights dont turn on with error 'h' format requires -32768 <= number <= 32767

```
Logger: homeassistant.components.websocket_api.http.connection
Source: custom_components/casambi_bt/light.py:230
Integration: Home Assistant WebSocket API (documentation, issues)
First occurred: 11:12:34 AM (10 occurrences)
Last logged: 11:16:49 AM

[547396467968] 'h' format requires -32768 <= number <= 32767
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 199, in handle_call_service
    await hass.services.async_call(
  File "/usr/src/homeassistant/homeassistant/core.py", line 1808, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1845, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 213, in handle_service
    await service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 686, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 961, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 726, in _handle_entity_call
    await result
  File "/usr/src/homeassistant/homeassistant/components/light/__init__.py", line 572, in async_handle_light_on_service
    await light.async_turn_on(**filter_turn_on_params(light, params))
  File "/config/custom_components/casambi_bt/light.py", line 230, in async_turn_on
    await self._api.casa.turnOn(self._obj)
  File "/usr/local/lib/python3.10/site-packages/CasambiBt/_casambi.py", line 246, in turnOn
    await self._send(target, b"\xff\x05", OpCode.SetLevel)
  File "/usr/local/lib/python3.10/site-packages/CasambiBt/_casambi.py", line 277, in _send
    opPkt = self._opContext.prepareOperation(opcode, targetCode, state)
  File "/usr/local/lib/python3.10/site-packages/CasambiBt/_operation.py", line 26, in prepareOperation
    packet = struct.pack(">hbhhh", flags, op, self.origin, target, 0)
struct.error: 'h' format requires -32768 <= number <= 32767

```

**Comment by lkempf** [comment]
Thanks. I would need full debug logging of the initial connection and of an attempt to turn on a light that isn't working to investigate this further. Please see https://github.com/lkempf/casambi-bt-hass#reporting-issues on how to enable this.

**Comment by rnd-rivertech** [comment]
Here it is, before and after attempt to turn on the light:

```
2023-04-04 13:10:27.800 DEBUG (MainThread) [CasambiBt._encryption] Decrypted package: b'069903134c04'
2023-04-04 13:10:27.802 DEBUG (MainThread) [CasambiBt._client] Incoming data of type 6: b'069903134c04'
2023-04-04 13:10:27.802 INFO (MainThread) [CasambiBt._client] Parsing incoming unit states...
2023-04-04 13:10:27.802 DEBUG (MainThread) [CasambiBt._client] Incoming unit state: b'9903134c04'
2023-04-04 13:10:27.802 DEBUG (MainThread) [CasambiBt._client] Parsed state: Id 153, prio 3, online True, on True, state b'4c04'1
2023-04-04 13:10:27.803 INFO (MainThread) [CasambiBt._casambi] Incomming data callback of type 6
2023-04-04 13:10:27.803 DEBUG (MainThread) [CasambiBt._casambi] Handling changed state b'4c04' for unit 153
2023-04-04 13:10:27.803 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 0: 0
2023-04-04 13:10:27.803 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 2: 275
2023-04-04 13:10:27.804 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 14: 0
2023-04-04 13:10:27.804 DEBUG (MainThread) [CasambiBt._unit] Parsed b'4c04' to UnitState(dimmer=None, vertical=None, rgb=None, white=None, temperature=None)
2023-04-04 13:10:29.835 DEBUG (MainThread) [CasambiBt._client] Callback on handle c9ffde48-ca5a-0001-ab83-8f519b482f77 (Handle: 9): Unknown: b'1a00008042888df9e504d3683ff000fb20c5402385b57e3a7ddc'
2023-04-04 13:10:29.836 INFO (MainThread) [CasambiBt._encryption] Decrypting packet: b'1a00008042888df9e504d3683ff000fb20c5402385b57e3a7ddc' of len 26 with nonce b'1a000080ff10a221d51067364ac9893c'
2023-04-04 13:10:29.838 DEBUG (MainThread) [CasambiBt._encryption] Decrypted package: b'069903135004'
2023-04-04 13:10:29.840 DEBUG (MainThread) [CasambiBt._client] Incoming data of type 6: b'069903135004'
2023-04-04 13:10:29.841 INFO (MainThread) [CasambiBt._client] Parsing incoming unit states...
2023-04-04 13:10:29.841 DEBUG (MainThread) [CasambiBt._client] Incoming unit state: b'9903135004'
2023-04-04 13:10:29.841 DEBUG (MainThread) [CasambiBt._client] Parsed state: Id 153, prio 3, online True, on True, state b'5004'1
2023-04-04 13:10:29.842 INFO (MainThread) [CasambiBt._casambi] Incomming data callback of type 6
2023-04-04 13:10:29.842 DEBUG (MainThread) [CasambiBt._casambi] Handling changed state b'5004' for unit 153
2023-04-04 13:10:29.842 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 0: 0
2023-04-04 13:10:29.842 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 2: 276
2023-04-04 13:10:29.843 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 14: 0
2023-04-04 13:10:29.843 DEBUG (MainThread) [CasambiBt._unit] Parsed b'5004' to UnitState(dimmer=None, vertical=None, rgb=None, white=None, temperature=None)
2023-04-04 13:10:32.806 DEBUG (MainThread) [CasambiBt._client] Callback on handle c9ffde48-ca5a-0001-ab83-8f519b482f77 (Handle: 9): Unknown: b'1b000080d9132854ffb3e01b26052f1d67df80d21f3e07246336'
2023-04-04 13:10:32.806 INFO (MainThread) [CasambiBt._encryption] Decrypting packet: b'1b000080d9132854ffb3e01b26052f1d67df80d21f3e07246336' of len 26 with nonce b'1b000080ff10a221d51067364ac9893c'
2023-04-04 13:10:32.809 DEBUG (MainThread) [CasambiBt._encryption] Decrypted package: b'069903135404'
2023-04-04 13:10:32.811 DEBUG (MainThread) [CasambiBt._client] Incoming data of type 6: b'069903135404'
2023-04-04 13:10:32.812 INFO (MainThread) [CasambiBt._client] Parsing incoming unit states...
2023-04-04 13:10:32.813 DEBUG (MainThread) [CasambiBt._client] Incoming unit state: b'9903135404'
2023-04-04 13:10:32.813 DEBUG (MainThread) [CasambiBt._client] Parsed state: Id 153, prio 3, online True, on True, state b'5404'1
2023-04-04 13:10:32.813 INFO (MainThread) [CasambiBt._casambi] Incomming data callback of type 6
2023-04-04 13:10:32.813 DEBUG (MainThread) [CasambiBt._casambi] Handling changed state b'5404' for unit 153
2023-04-04 13:10:32.814 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 0: 0
2023-04-04 13:10:32.814 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 2: 277
2023-04-04 13:10:32.814 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 14: 0
2023-04-04 13:10:32.814 DEBUG (MainThread) [CasambiBt._unit] Parsed b'5404' to UnitState(dimmer=None, vertical=None, rgb=None, white=None, temperature=None)
2023-04-04 13:10:34.859 DEBUG (MainThread) [CasambiBt._client] Callback on handle c9ffde48-ca5a-0001-ab83-8f519b482f77 (Handle: 9): Unknown: b'1c0000803fa2f4782f24ec864598938d3b8fbccbf0c1fb966f33'
2023-04-04 13:10:34.859 INFO (MainThread) [CasambiBt._encryption] Decrypting packet: b'1c0000803fa2f4782f24ec864598938d3b8fbccbf0c1fb966f33' of len 26 with nonce b'1c000080ff10a221d51067364ac9893c'
2023-04-04 13:10:34.862 DEBUG (MainThread) [CasambiBt._encryption] Decrypted package: b'069903135804'
2023-04-04 13:10:34.864 DEBUG (MainThread) [CasambiBt._client] Incoming data of type 6: b'069903135804'
2023-04-04 13:10:34.864 INFO (MainThread) [CasambiBt._client] Parsing incoming unit states...
2023-04-04 13:10:34.865 DEBUG (MainThread) [CasambiBt._client] Incoming unit state: b'9903135804'
2023-04-04 13:10:34.865 DEBUG (MainThread) [CasambiBt._client] Parsed state: Id 153, prio 3, online True, on True, state b'5804'1
2023-04-04 13:10:34.865 INFO (MainThread) [CasambiBt._casambi] Incomming data callback of type 6
2023-04-04 13:10:34.865 DEBUG (MainThread) [CasambiBt._casambi] Handling changed state b'5804' for unit 153
2023-04-04 13:10:34.866 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 0: 0
2023-04-04 13:10:34.866 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 2: 278
2023-04-04 13:10:34.866 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 14: 0
2023-04-04 13:10:34.866 DEBUG (MainThread) [CasambiBt._unit] Parsed b'5804' to UnitState(dimmer=None, vertical=None, rgb=None, white=None, temperature=None)
2023-04-04 13:10:36.510 DEBUG (MainThread) [CasambiBt._casambi] Sending operation SetLevel with payload b'ff05' for 9801
2023-04-04 13:10:36.511 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [547882078832] 'h' format requires -32768 <= number <= 32767
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 199, in handle_call_service
    await hass.services.async_call(
  File "/usr/src/homeassistant/homeassistant/core.py", line 1808, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1845, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 213, in handle_service
    await service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 686, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 961, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 726, in _handle_entity_call
    await result
  File "/usr/src/homeassistant/homeassistant/components/light/__init__.py", line 572, in async_handle_light_on_service
    await light.async_turn_on(**filter_turn_on_params(light, params))
  File "/config/custom_components/casambi_bt/light.py", line 230, in async_turn_on
    await self._api.casa.turnOn(self._obj)
  File "/usr/local/lib/python3.10/site-packages/CasambiBt/_casambi.py", line 246, in turnOn
    await self._send(target, b"\xff\x05", OpCode.SetLevel)
  File "/usr/local/lib/python3.10/site-packages/CasambiBt/_casambi.py", line 277, in _send
    opPkt = self._opContext.prepareOperation(opcode, targetCode, state)
  File "/usr/local/lib/python3.10/site-packages/CasambiBt/_operation.py", line 26, in prepareOperation
    packet = struct.pack(">hbhhh", flags, op, self.origin, target, 0)
struct.error: 'h' format requires -32768 <= number <= 32767
2023-04-04 13:10:36.714 DEBUG (MainThread) [CasambiBt._client] Callback on handle c9ffde48-ca5a-0001-ab83-8f519b482f77 (Handle: 9): Unknown: b'1d000080c42a0317b9db393fefabf1eaca4a95ff152d3ae478ca'
2023-04-04 13:10:36.714 INFO (MainThread) [CasambiBt._encryption] Decrypting packet: b'1d000080c42a0317b9db393fefabf1eaca4a95ff152d3ae478ca' of len 26 with nonce b'1d000080ff10a221d51067364ac9893c'
2023-04-04 13:10:36.717 DEBUG (MainThread) [CasambiBt._encryption] Decrypted package: b'069903135c04'
2023-04-04 13:10:36.718 DEBUG (MainThread) [CasambiBt._client] Incoming data of type 6: b'069903135c04'
2023-04-04 13:10:36.719 INFO (MainThread) [CasambiBt._client] Parsing incoming unit states...
2023-04-04 13:10:36.719 DEBUG (MainThread) [CasambiBt._client] Incoming unit state: b'9903135c04'
2023-04-04 13:10:36.719 DEBUG (MainThread) [CasambiBt._client] Parsed state: Id 153, prio 3, online True, on True, state b'5c04'1
2023-04-04 13:10:36.719 INFO (MainThread) [CasambiBt._casambi] Incomming data callback of type 6
2023-04-04 13:10:36.719 DEBUG (MainThread) [CasambiBt._casambi] Handling changed state b'5c04' for unit 153
2023-04-04 13:10:36.720 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 0: 0
2023-04-04 13:10:36.720 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 2: 279
2023-04-04 13:10:36.720 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 14: 0
2023-04-04 13:10:36.720 DEBUG (MainThread) [CasambiBt._unit] Parsed b'5c04' to UnitState(dimmer=None, vertical=None, rgb=None, white=None, temperature=None)
2023-04-04 13:10:38.759 DEBUG (MainThread) [CasambiBt._client] Callback on handle c9ffde48-ca5a-0001-ab83-8f519b482f77 (Handle: 9): Unknown: b'1e000080a9ac5f98f85ba2744fc9a529a2628ef54911d89b6262'
2023-04-04 13:10:38.760 INFO (MainThread) [CasambiBt._encryption] Decrypting packet: b'1e000080a9ac5f98f85ba2744fc9a529a2628ef54911d89b6262' of len 26 with nonce b'1e000080ff10a221d51067364ac9893c'
2023-04-04 13:10:38.765 DEBUG (MainThread) [CasambiBt._encryption] Decrypted package: b'069903136004'
2023-04-04 13:10:38.768 DEBUG (MainThread) [CasambiBt._client] Incoming data of type 6: b'069903136004'
2023-04-04 13:10:38.768 INFO (MainThread) [CasambiBt._client] Parsing incoming unit states...
2023-04-04 13:10:38.768 DEBUG (MainThread) [CasambiBt._client] Incoming unit state: b'9903136004'
2023-04-04 13:10:38.769 DEBUG (MainThread) [CasambiBt._client] Parsed state: Id 153, prio 3, online True, on True, state b'6004'1
2023-04-04 13:10:38.769 INFO (MainThread) [CasambiBt._casambi] Incomming data callback of type 6
2023-04-04 13:10:38.769 DEBUG (MainThread) [CasambiBt._casambi] Handling changed state b'6004' for unit 153
2023-04-04 13:10:38.770 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 0: 0
2023-04-04 13:10:38.770 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 2: 280
2023-04-04 13:10:38.770 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 14: 0
2023-04-04 13:10:38.770 DEBUG (MainThread) [CasambiBt._unit] Parsed b'6004' to UnitState(dimmer=None, vertical=None, rgb=None, white=None, temperature=None)
2023-04-04 13:10:39.799 DEBUG (MainThread) [CasambiBt._client] Callback on handle c9ffde48-ca5a-0001-ab83-8f519b482f77 (Handle: 9): Unknown: b'1f000080aa27c54d1c6a601ab0d02e779a3eb3c510ccf365390e'
2023-04-04 13:10:39.799 INFO (MainThread) [CasambiBt._encryption] Decrypting packet: b'1f000080aa27c54d1c6a601ab0d02e779a3eb3c510ccf365390e' of len 26 with nonce b'1f000080ff10a221d51067364ac9893c'
2023-04-04 13:10:39.800 DEBUG (MainThread) [CasambiBt._encryption] Decrypted package: b'06940313c401'
2023-04-04 13:10:39.802 DEBUG (MainThread) [CasambiBt._client] Incoming data of type 6: b'06940313c401'
2023-04-04 13:10:39.804 INFO (MainThread) [CasambiBt._client] Parsing incoming unit states...
2023-04-04 13:10:39.804 DEBUG (MainThread) [CasambiBt._client] Incoming unit state: b'940313c401'
2023-04-04 13:10:39.804 DEBUG (MainThread) [CasambiBt._client] Parsed state: Id 148, prio 3, online True, on True, state b'c401'1
2023-04-04 13:10:39.804 INFO (MainThread) [CasambiBt._casambi] Incomming data callback of type 6
2023-04-04 13:10:39.804 DEBUG (MainThread) [CasambiBt._casambi] Handling changed state b'c401' for unit 148
2023-04-04 13:10:39.805 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 0: 0
2023-04-04 13:10:39.805 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 2: 113
2023-04-04 13:10:39.805 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 14: 0
2023-04-04 13:10:39.805 DEBUG (MainThread) [CasambiBt._unit] Parsed b'c401' to UnitState(dimmer=None, vertical=None, rgb=None, white=None, temperature=None)
2023-04-04 13:10:40.705 DEBUG (MainThread) [CasambiBt._client] Callback on handle c9ffde48-ca5a-0001-ab83-8f519b482f77 (Handle: 9): Unknown: b'200000806c6ae29a5bdd704595c573738141a50ca8e5ad448206'
2023-04-04 13:10:40.705 INFO (MainThread) [CasambiBt._encryption] Decrypting packet: b'200000806c6ae29a5bdd704595c573738141a50ca8e5ad448206' of len 26 with nonce b'20000080ff10a221d51067364ac9893c'
2023-04-04 13:10:40.707 DEBUG (MainThread) [CasambiBt._encryption] Decrypted package: b'069903136404'
2023-04-04 13:10:40.712 DEBUG (MainThread) [CasambiBt._client] Incoming data of type 6: b'069903136404'
2023-04-04 13:10:40.712 INFO (MainThread) [CasambiBt._client] Parsing incoming unit states...
2023-04-04 13:10:40.713 DEBUG (MainThread) [CasambiBt._client] Incoming unit state: b'9903136404'
2023-04-04 13:10:40.713 DEBUG (MainThread) [CasambiBt._client] Parsed state: Id 153, prio 3, online True, on True, state b'6404'1
2023-04-04 13:10:40.713 INFO (MainThread) [CasambiBt._casambi] Incomming data callback of type 6
2023-04-04 13:10:40.714 DEBUG (MainThread) [CasambiBt._casambi] Handling changed state b'6404' for unit 153
2023-04-04 13:10:40.714 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 0: 0
2023-04-04 13:10:40.714 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 2: 281
2023-04-04 13:10:40.714 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 14: 0
2023-04-04 13:10:40.715 DEBUG (MainThread) [CasambiBt._unit] Parsed b'6404' to UnitState(dimmer=None, vertical=None, rgb=None, white=None, temperature=None)
2023-04-04 13:10:41.708 DEBUG (MainThread) [CasambiBt._client] Callback on handle c9ffde48-ca5a-0001-ab83-8f519b482f77 (Handle: 9): Unknown: b'210000806bd4ce0c51672721e2e66124c0da2d2d44996240b110'
2023-04-04 13:10:41.709 INFO (MainThread) [CasambiBt._encryption] Decrypting packet: b'210000806bd4ce0c51672721e2e66124c0da2d2d44996240b110' of len 26 with nonce b'21000080ff10a221d51067364ac9893c'
2023-04-04 13:10:41.712 DEBUG (MainThread) [CasambiBt._encryption] Decrypted package: b'069903136804'
2023-04-04 13:10:41.714 DEBUG (MainThread) [CasambiBt._client] Incoming data of type 6: b'069903136804'
2023-04-04 13:10:41.714 INFO (MainThread) [CasambiBt._client] Parsing incoming unit states...
2023-04-04 13:10:41.715 DEBUG (MainThread) [CasambiBt._client] Incoming unit state: b'9903136804'
2023-04-04 13:10:41.715 DEBUG (MainThread) [CasambiBt._client] Parsed state: Id 153, prio 3, online True, on True, state b'6804'1
2023-04-04 13:10:41.715 INFO (MainThread) [CasambiBt._casambi] Incomming data callback of type 6
2023-04-04 13:10:41.715 DEBUG (MainThread) [CasambiBt._casambi] Handling changed state b'6804' for unit 153
2023-04-04 13:10:41.715 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 0: 0
2023-04-04 13:10:41.716 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 2: 282
2023-04-04 13:10:41.716 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 14: 0
2023-04-04 13:10:41.716 DEBUG (MainThread) [CasambiBt._unit] Parsed b'6804' to UnitState(dimmer=None, vertical=None, rgb=None, white=None, temperature=None)
2023-04-04 13:10:43.593 DEBUG (MainThread) [CasambiBt._client] Callback on handle c9ffde48-ca5a-0001-ab83-8f519b482f77 (Handle: 9): Unknown: b'22000080e9c2f58783172b531bfe66b95232adeeec30245e6e0d'
2023-04-04 13:10:43.594 INFO (MainThread) [CasambiBt._encryption] Decrypting packet: b'22000080e9c2f58783172b531bfe66b95232adeeec30245e6e0d' of len 26 with nonce b'22000080ff10a221d51067364ac9893c'
2023-04-04 13:10:43.595 DEBUG (MainThread) [CasambiBt._encryption] Decrypted package: b'06940313c801'
2023-04-04 13:10:43.600 DEBUG (MainThread) [CasambiBt._client] Incoming data of type 6: b'06940313c801'
2023-04-04 13:10:43.601 INFO (MainThread) [CasambiBt._client] Parsing incoming unit states...
2023-04-04 13:10:43.601 DEBUG (MainThread) [CasambiBt._client] Incoming unit state: b'940313c801'
2023-04-04 13:10:43.601 DEBUG (MainThread) [CasambiBt._client] Parsed state: Id 148, prio 3, online True, on True, state b'c801'1
2023-04-04 13:10:43.602 INFO (MainThread) [CasambiBt._casambi] Incomming data callback of type 6
2023-04-04 13:10:43.602 DEBUG (MainThread) [CasambiBt._casambi] Handling changed state b'c801' for unit 148
2023-04-04 13:10:43.602 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 0: 0
2023-04-04 13:10:43.602 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 2: 114
2023-04-04 13:10:43.603 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 14: 0
2023-04-04 13:10:43.603 DEBUG (MainThread) [CasambiBt._unit] Parsed b'c801' to UnitState(dimmer=None, vertical=None, rgb=None, white=None, temperature=None)
2023-04-04 13:10:43.777 DEBUG (MainThread) [CasambiBt._client] Callback on handle c9ffde48-ca5a-0001-ab83-8f519b482f77 (Handle: 9): Unknown: b'23000080634b6a0f3209b857aac262c55d48e93af76b9888ea62'
2023-04-04 13:10:43.778 INFO (MainThread) [CasambiBt._encryption] Decrypting packet: b'23000080634b6a0f3209b857aac262c55d48e93af76b9888ea62' of len 26 with nonce b'23000080ff10a221d51067364ac9893c'
2023-04-04 13:10:43.780 DEBUG (MainThread) [CasambiBt._encryption] Decrypted package: b'069903136c04'
2023-04-04 13:10:43.782 DEBUG (MainThread) [CasambiBt._client] Incoming data of type 6: b'069903136c04'
2023-04-04 13:10:43.783 INFO (MainThread) [CasambiBt._client] Parsing incoming unit states...
2023-04-04 13:10:43.783 DEBUG (MainThread) [CasambiBt._client] Incoming unit state: b'9903136c04'
2023-04-04 13:10:43.784 DEBUG (MainThread) [CasambiBt._client] Parsed state: Id 153, prio 3, online True, on True, state b'6c04'1
2023-04-04 13:10:43.784 INFO (MainThread) [CasambiBt._casambi] Incomming data callback of type 6
2023-04-04 13:10:43.784 DEBUG (MainThread) [CasambiBt._casambi] Handling changed state b'6c04' for unit 153
2023-04-04 13:10:43.785 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 0: 0
2023-04-04 13:10:43.785 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 2: 283
2023-04-04 13:10:43.785 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 14: 0
2023-04-04 13:10:43.785 DEBUG (MainThread) [CasambiBt._unit] Parsed b'6c04' to UnitState(dimmer=None, vertical=None, rgb=None, white=None, temperature=None)
2023-04-04 13:10:46.825 DEBUG (MainThread) [CasambiBt._client] Callback on handle c9ffde48-ca5a-0001-ab83-8f519b482f77 (Handle: 9): Unknown: b'240000807d5e717f58d23515a00d8904e624fc2049c4908ea53c'
2023-04-04 13:10:46.826 INFO (MainThread) [CasambiBt._encryption] Decrypting packet: b'240000807d5e717f58d23515a00d8904e624fc2049c4908ea53c' of len 26 with nonce b'24000080ff10a221d51067364ac9893c'
2023-04-04 13:10:46.832 DEBUG (MainThread) [CasambiBt._encryption] Decrypted package: b'069903137004'
2023-04-04 13:10:46.835 DEBUG (MainThread) [CasambiBt._client] Incoming data of type 6: b'069903137004'
2023-04-04 13:10:46.836 INFO (MainThread) [CasambiBt._client] Parsing incoming unit states...
2023-04-04 13:10:46.836 DEBUG (MainThread) [CasambiBt._client] Incoming unit state: b'9903137004'
2023-04-04 13:10:46.836 DEBUG (MainThread) [CasambiBt._client] Parsed state: Id 153, prio 3, online True, on True, state b'7004'1
2023-04-04 13:10:46.836 INFO (MainThread) [CasambiBt._casambi] Incomming data callback of type 6
2023-04-04 13:10:46.836 DEBUG (MainThread) [CasambiBt._casambi] Handling changed state b'7004' for unit 153
2023-04-04 13:10:46.837 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 0: 0
2023-04-04 13:10:46.837 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 2: 284
2023-04-04 13:10:46.837 DEBUG (MainThread) [CasambiBt._unit] Value for unkown control type at 14: 0
2023-04-04 13:10:46.837 DEBUG (MainThread) [CasambiBt._unit] Parsed b'7004' to UnitState(dimmer=None, vertical=None, rgb=None, white=None, temperature=None)
```

**Comment by harisma-git** [comment]
Is it releted that is 0-10V dimmer? model CBU-ASD (0/1-10)

**Comment by lkempf** [comment]
That's still missing the log of the initial connection. I need that to find the unit type.

**Comment by lkempf** [comment]
I had a look at the unit type I got from the log @harisma-git sent me and I have no idea how this error can happen. I've tried to implement a workaround for version 0.0.6 and hope that this fixes it. If not I will have to add more logging.

**Comment by harisma-git** [comment]
Unfortunately 0.0.6 doesnt solve the problem (same error). I dont know if the problem is that these devices are 0-10V dimmers in the meaning that they dim every 10%. 0V for off, 1V for 10%, 2V for 20% and so on. In cloud integration working fine.

**Comment by lkempf** [comment]
The underlying issue was that the id of the unit was larger than 127. Fixed in 0.0.7 (unless there are lights with ids > 255 but then the fix won't be as easy).
